### PR TITLE
Chore run tests on common browsers

### PR DESCRIPTION
**Pull request by faloi**
# :dart: Goal

Run Capybara and Teaspoon tests on the three common browsers: Chrome, Firefox and Safari.

# :memo: Details

* Browser is selected by settings `MUMUKI_SELENIUM_DRIVER` env variable. 
* All the web drivers are now installed by the [webdrivers gem](https://github.com/titusfortner/webdrivers).
* Migrated Travis configuration to a multi-job build and updated it to remove obsolete attributes.

# :soon: Future work

Unfortunately, some tests were skipped because they didn't work out of the box with the Selenium driver. We should further investigate if they can be fixed or not.

Also, some tests break on Safari due to user session issues. Until this is fixed, Safari builds on Travis are allowed to fail.

# :eyes: See also 

* #1486
* #1492
* #1479

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -25,14 +25,18 @@ require 'rspec/core'
 require 'rspec/core/rake_task'
 
 desc "Run all specs in spec directory (excluding plugin specs)"
-RSpec::Core::RakeTask.new(:spec => 'app:db:test:prepare')
```
Comment (faloi) [inline Rakefile:28]: This caused concurrency issues and I couldn't notice any difference after removing it.

File: Gemfile [hunk 2]
```diff
@@ -14,7 +14,7 @@ group :test do
   gem 'factory_bot_rails', '~> 5.0'
   gem 'faker', '~> 2.2'
   gem 'rake', '~> 12.3'
-  gem 'capybara', '~> 2.3.0'
+  gem 'capybara', '~> 3.33.0'
```
Comment (faloi) [inline Gemfile:17]: Some of the headless drivers were introduced in version 3. Only a few tests broke after the upgrade.

**Review by faloi — COMMENTED** [review]
Some comments to clarify.

File: .travis.yml [hunk 3]
```diff
@@ -1,34 +1,79 @@
...
-    global:
-      - MOZ_HEADLESS=1
+- xvfb
+_hosts: &hosts
+  hosts:
+    - localmumuki.io
+    - base.localmumuki.io
+    - central.localmumuki.io
+    - empty.localmumuki.io
+    - foo.localmumuki.io
```
Comment (flbulgarelli) [inline .travis.yml:26]: @luchotc we need to remove these tests or at least convert most of them to path mapping, which is the recommended one. Also, we should remove localmumuki which is a implicit mapping. See https://github.com/mumuki/mumuki-laboratory/issues/1492

File: .travis.yml [hunk 4]
```diff
@@ -1,34 +1,79 @@
...
+        - bundle exec rake capybara
+        - bundle exec rake teaspoon
+      addons:
+        chrome: stable
+        <<: *hosts
+
+    # Capybara and Teaspoon tests on Firefox
+    - env: MUMUKI_SELENIUM_DRIVER=firefox
+      script:
+        - bundle exec rake capybara
```
Comment (flbulgarelli) [inline .travis.yml:52]: Perhaps we could create a new task for running capybara + teaspoon, like `rake test:web` or something like that

File: README.md [hunk 5]
```diff
@@ -135,13 +135,28 @@ rails s
 bundle exec rspec
 ```
 
+## Running Capybara tests with Selenium
+
+The Capybara config of this project supports running tests on Firefox, Chrome and Safari via Selenium. The [`webdrivers`](https://github.com/titusfortner/webdrivers) gem automatically installs (and updates) all the necessary Selenium webdrivers.
```
Comment (flbulgarelli) [inline README.md:140]: :heart:

File: spec/capybara_helper.rb [hunk 6]
```diff
@@ -10,3 +25,86 @@ def set_implicit_central!
...
+
+    class ActionDispatch::Request
+      prepend RequestWithExtraHeaders
+    end
+  EOR
+end
+
+def exclude_selenium_failing_tests!
+  RSpec.configure do |config|
+    config.filter_run_excluding(
```
Comment (flbulgarelli) [inline spec/capybara_helper.rb:79]: Does rspec when these tests are ignored?

**Review by flbulgarelli — APPROVED** [review]

File: .travis.yml (hunk 4 above)
Comment (faloi) [reply .travis.yml:52]: Agree. Perhaps `spec:web`, composed of `spec:web:capybara` and `spec:web:teaspoon`.

File: spec/capybara_helper.rb (hunk 6 above)
Comment (faloi) [reply spec/capybara_helper.rb:79]: By default, it outputs the following info when `filter_run_excluding` is used:

```
Run options: exclude {:xpath_no_matches=>true, :organization_not_nil=>true, :navigation_error=>true, :json_eq_error=>true, :invalid_selector_error=>true, :element_not_interactable_error=>true, :http_response_headers=>true}
```

On top of that, I've added another line to be sure it's running with the desired Selenium driver:

```
Running Capybara tests with selenium_headless, not ignoring hidden elements`
```

**Comment by flbulgarelli** [comment]
It is broken now :(

**Review by flbulgarelli — APPROVED** [review]

**Review by luchotc — APPROVED** [review]
This is GREAT @faloi :clap:

**Pull request merged**